### PR TITLE
feat: add get team staff workflow

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** to support teams in ESG, add the ability to get a staff workflow for a team.

**Note:** based on the way the tickets are broken up, nothing exercises this yet so there are no unit tests yet. I did add the mock context manager though for future test use.

JIRA: [AU-495](https://openedx.atlassian.net/browse/AU-495)

**What changed?**

- add `TeamStaffWorkflow.get_team_staff_workflow`
- add mock context manager for the above.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/edx/edx-ora2/blob/master/.github/release_process.md)
- [ ] Bumped version number in [setup.py](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/edx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

n/a

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/staffgrader/tests/test_get_assessment_info.py [hunk 1]
```diff
@@ -42,6 +42,15 @@ def _mock_get_staff_workflow(self, **kwargs):
         ) as mock_get:
             yield mock_get
 
+    @contextmanager
+    def _mock_get_team_staff_workflow(self, **kwargs):
```
Comment (jansenk) [inline openassessment/staffgrader/tests/test_get_assessment_info.py:46]: Is this not used?

**Review by jansenk — APPROVED** [review]
lgtm but i don't know that we need the mock if we aren't using it
